Sources/Sentry/SentryTracer.m

```diff
@@ -689,10 +691,42 @@ - (SentryTransaction *)toTransaction
 
     SentryTransaction *transaction = [[SentryTransaction alloc] initWithTrace:self children:spans];
     transaction.transaction = self.transactionContext.name;
+
+    if (self.requiresDebugMeta) {
+        SentryDebugImageProvider *debugImageProvider
+            = SentryDependencyContainer.sharedInstance.debugImageProvider;
+        NSMutableSet<NSString *> *imageAddresses = [NSMutableSet set];
+        [imageAddresses addObjectsFromArray:[self extractFramesImageAddressFromSpan:self.rootSpan]];
+        for (id<SentrySpan> span in _children) {
+            [imageAddresses addObjectsFromArray:[self extractFramesImageAddressFromSpan:span]];
+        }
+        transaction.debugMeta =
+            [debugImageProvider getDebugImagesForAddresses:imageAddresses.allObjects];
+    }
+
     [self addMeasurements:transaction];
     return transaction;
 }
 
+- (NSArray<NSString *> *)extractFramesImageAddressFromSpan:(id<SentrySpan>)span
+{
+    NSArray *spanFrames = span.data[@"call_stack"];
+    NSMutableArray<NSString *> *result = [NSMutableArray array];
+    if (spanFrames && [spanFrames isKindOfClass:NSArray.class]) {
+        for (id object in spanFrames) {
+            if ([object isKindOfClass:NSDictionary.class] && object[@"image_addr"]) {
+                [result addObject:object[@"image_addr"]];
+            }
+        }
+    }
+    return result;
+}
```

Comment:
`m`: As the tracer already contains so much code, what do you think about extracting some of that code into another class? Something like
```ObjC
if (self.requiresDebugMeta) {
    transaction.debugMeta = [transactionDebugMetaProvider getDebugMetaForSpan:self.rootSpan];
}
```